### PR TITLE
Fix/subgraph apr

### PR DESCRIPTION
**Pull request by ixm-759**

### CODE REVIEW

File: src/stats/subgraph.service.ts [hunk 1]
```diff
@@ -163,4 +179,252 @@ export class SubgraphService {
...
+    try {
+      const { data } = await this.httpService.post(url, { query }).toPromise();
+      return data;
+    } catch (error) {
+      console.log('error');
+      console.log(error);
+    }
+  }
+
+  async getEthPrice(chainId) {
```
Comment (ape-swap) [inline src/stats/subgraph.service.ts:193]: Could you typesafe this function and use a more semantic name? 

Even though the Subgraph calls the protocol-token ETH. Neither of our chains uses ETH.

File: src/stats/subgraph.service.ts [hunk 2]
```diff
@@ -163,4 +179,252 @@ export class SubgraphService {
...
+      const currentPrice = result?.data?.bundles[0]?.ethPrice;
+      const oneDayBackPrice = resultOneDay?.data?.bundles[0]?.ethPrice;
+      priceChangeETH = getPercentChange(currentPrice, oneDayBackPrice);
+      ethPrice = currentPrice;
+      ethPriceOneDay = oneDayBackPrice;
+    } catch (e) {
+      console.log(e);
+    }
+
+    return [ethPrice, ethPriceOneDay, priceChangeETH];
```
Comment (ape-swap) [inline src/stats/subgraph.service.ts:225]: Please. Return a typesafed object with semantic names on their properties.

This approach is not recommended and can easily grow out of control.

File: src/stats/subgraph.service.ts [hunk 3]
```diff
@@ -163,4 +179,252 @@ export class SubgraphService {
...
+  async getBlockFromTimestamp(timestamp: number, chainId) {
+    const url = this.configService.getData<string>(
+      `${chainId}.subgraph.blocks`,
+    );
+    const result = await this.executeQuerySubraph(
+      url,
+      GET_BLOCK(timestamp, timestamp + 600),
+    );
+    return result?.data?.blocks?.[0]?.number;
+  }
```
Comment (ape-swap) [inline src/stats/subgraph.service.ts:367]: I this needed or can we call `getBlocksFromTimestamps` with one timestamp?

File: src/stats/subgraph.service.ts [hunk 4]
```diff
@@ -163,4 +179,252 @@ export class SubgraphService {
...
+    }
+    if (!oneDayData && data) {
+      data.oneDayVolumeUSD = parseFloat(data.volumeUSD);
+    }
+    if (!oneWeekData && data) {
+      data.oneWeekVolumeUSD = parseFloat(data.volumeUSD);
+    }
+
+    return data;
+  }
```
Comment (ape-swap) [inline src/stats/subgraph.service.ts:356]: Is there a way to compartmentalize this into smaller chunks of more readeable code?

I would consider splitting some of this code on another file, relative to the responsabilities of the functions.

File: src/utils/helpers.ts [hunk 5]
```diff
@@ -31,3 +33,42 @@ export function createLpPairName(t0, t1) {
...
     : `[${t1}]-[${t0}] LP`;
 }
+
+export function getTimestampsForChanges() {
+  const utcCurrentTime = dayjs();
+  const t1 = utcCurrentTime.subtract(1, 'day').startOf('minute').unix();
+  const t2 = utcCurrentTime.subtract(2, 'day').startOf('minute').unix();
+  const tWeek = utcCurrentTime.subtract(1, 'week').startOf('minute').unix();
+  return [t1, t2, tWeek];
+}
```
Comment (ape-swap) [inline src/utils/helpers.ts:43]: Please typesafe this function and return a semantic object instead of an array.

File: src/utils/helpers.ts [hunk 6]
```diff
@@ -31,3 +33,42 @@ export function createLpPairName(t0, t1) {
...
+    parseFloat(value24HoursAgo) - parseFloat(value48HoursAgo);
+
+  const adjustedPercentChange =
+    (currentChange - previousChange / previousChange) * 100;
+
+  if (isNaN(adjustedPercentChange) || !isFinite(adjustedPercentChange)) {
+    return [currentChange, 0];
+  }
+  return [currentChange, adjustedPercentChange];
+};
```
Comment (ape-swap) [inline src/utils/helpers.ts:74]: Please return typesafed semantic objects instead of arrays in instances like this.

**Review by ape-swap — CHANGES_REQUESTED** [review]
- Please typesafe every function by default.
- Please when applicable return semantic object structure from functions, not arrays.
- Would be great to refactor the `subgraph.service.ts` file into smaller chunk functions, create a separate file or both. 

Let me know if you have any questions or concerns.

File: src/stats/subgraph.service.ts (hunk 1 above)
Comment (ixm-759) [reply src/stats/subgraph.service.ts]: Done!

File: src/stats/subgraph.service.ts (hunk 2 above)
Comment (ixm-759) [reply src/stats/subgraph.service.ts]: Done!

File: src/stats/subgraph.service.ts (hunk 3 above)
Comment (ixm-759) [reply src/stats/subgraph.service.ts]: I eliminated this method and used another one that was already there but in bulk, I adapted it to the method I was using.

File: src/stats/subgraph.service.ts (hunk 4 above)
Comment (ixm-759) [reply src/stats/subgraph.service.ts]: I split the method, making other methods more readable and easier to read.

File: src/utils/helpers.ts (hunk 5 above)
Comment (ixm-759) [reply src/utils/helpers.ts]: Done!

File: src/utils/helpers.ts (hunk 6 above)
Comment (ixm-759) [reply src/utils/helpers.ts]: Done!

**Review by ape-swap — APPROVED** [review]
Nice refactor! Thanks

**Pull request merged**